### ISSUE TITLE
error TS2305: Module '"entities/lib/decode.js"' has no exported member 'EntityDecoder'.

### ISSUE THREAD

**Issue by shashi-chemist2u** [inikulin/parse5#1296]
node_modules/parse5/dist/tokenizer/index.d.ts:3:10 - error TS2305: Module '"entities/lib/decode.js"' has no exported member 'EntityDecoder'.

3 import { EntityDecoder } from 'entities/lib/decode.js';
~~~~~~~~~~~~~

Found 1 error in node_modules/parse5/dist/tokenizer/index.d.ts:3

parse5@7.2.0 version has breaking change was fine on parse5@7.1.2 version

**Comment by 43081j** [comment]
Interesting. entities has exported `EntityDecoder` for a while now

@fb55 any idea about this? Maybe npm needs to bump the locally installed entities version but we did bad semver? I don't see how though 🤔

**Comment by cardoso** [comment]
Can I help with this? I believe we’re running into this issue in https://github.com/salesforce/lwc if we try to upgrade. It could be an issue on our own dependency graph though, I didn’t have enough time to rule that out 100%. Being stuck on yarn v1 is no fun for figuring this sort of thing out :(

**Comment by cardoso** [comment]
@43081j yes, latest. The issue is that I would like to update the entities package to latest (v6), but `parse5` is brings in v4:

```
parse5@^7.0.0, parse5@^7.1.2, parse5@^7.2.1:
  version "7.2.1"
  resolved "https://registry.yarnpkg.com/parse5/-/parse5-7.2.1.tgz#8928f55915e6125f430cc44309765bf17556a33a"
  integrity sha512-BuBYQYlv1ckiPdQi/ohiivi9Sagc9JG+Ozs0r7b/0iK3sKmrb0b9FdWdBbOdx6hBCM/F9Ir82ofnBhtZOjCRPQ==
  dependencies:
    entities "^4.5.0"
```

Looking at the [release notes of entities](https://github.com/fb55/entities/releases) v5 and v6, this would require some changes. Example:

https://github.com/inikulin/parse5/blob/b88c93886d28b9552fda9e472175c99d4c76ab9b/packages/parse5/lib/serializer/index.ts#L2

Can I submit a PR attempting to upgrade parse5 to use entities v6? No rush in reviewing/merging/releasing.

**Comment by cardoso** [comment]
Actually it seems @fb55 was attempting that here: https://github.com/inikulin/parse5/pull/1202

I see it caused some issues in CI, which I'd also be willing to solve.

**Comment by 43081j** [comment]
ah i understand now

i hadn't realised there's a new version of `entities` beyond 4.5.x

yes we basically need to get #1202 over the line then 👍  if you want to have a look at that, feel free. its a dependabot PR (automated), so i'd just check a new branch out from that one and do your changes on top

**Comment by cardoso** [comment]
@43081j I pushed https://github.com/inikulin/parse5/pull/1410 , tests are passing locally, but I'm not sure what to do about the cjs build yet. Is it something that can be dropped in a major version? Or should I look into a way of preserving it?
